src/nupic/math/ArrayAlgo.hpp

```diff
@@ -5241,7 +5142,7 @@ namespace nupic {
                      : 
                      );
 
-  #elif defined(NTA_ARCH_64)
+  #else //64bit
```

Comment:
Isn't it better to leave the block explicit on 64 bit? If we have some random other platform it would be better for it to fail then to used some block not intended for the architecture.